controllers/controllers/client.go

```diff
@@ -0,0 +1,196 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+
+	anywhere "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	spec "github.com/aws/eks-anywhere/pkg/cluster"
+	anywhererelease "github.com/aws/eks-anywhere/release/api/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capicontrolplane "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *ClusterReconcilerV2) BuildRemoteClientForCluster(ctx context.Context, cluster *anywhere.Cluster) (*eksaClient, error) {
+	capiCluster, err := r.Client.GetCAPICluster(ctx, cluster)
+	if err != nil {
+		return nil, err
+	}
+	key := client.ObjectKey{
+		Namespace: capiCluster.Namespace,
+		Name:      capiCluster.Name,
+	}
+	remoteClient, err := r.Tracker.GetClient(ctx, key)
+	if err != nil {
+		return nil, err
+	}
+
+	return &eksaClient{Client: remoteClient}, nil
+}
+
+type eksaClient struct {
+	client.Client
+}
+
+func (c eksaClient) BuildClusterSpec(ctx context.Context, cluster *anywhere.Cluster) (*cluster.Spec, error) {
+	bundles, err := c.GetBundlesForCluster(ctx, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	return spec.BuildSpecFromBundles(cluster, bundles)
+}
+
+func (c eksaClient) GetBundlesForCluster(ctx context.Context, cluster *anywhere.Cluster) (*anywhererelease.Bundles, error) {
+	// TODO: the bundles name can be cached in the status
+	// TODO: answer this question: do the management cluster and all the workload clusters use the same Bundle?
+	//  if they don't, that opens a lot our testing matrix. Right now they do and they don't. The cli is not enforcing it,
+	//  but the v0.6.0 is the only cli that supports workload clusters and there is only one bundle compatible with it
+	// For now, I'm gonna get the bundles using the cluster name, but we should really change that
+
+	// Once we have the bundles ref in the status, we can add a check here and use it if present
+	// If the status is not there, it means it's a new cluster, so we get the bundles being used by the management cluster and use it
+	// Maybe using the spec would be better than the status. Status is supposed to be recoverable from the state of the cluster,
+	// and this seems more ephemeral
+	// TODO: This assumed the Bundles object has the same name as the management cluster. Change when we have a ref
+	// TODO: fix namespace issue, it seems like we install tge Bundles in the default namespace
```

Comment:
i think we install all eksa resources in default namespace

Reply:
I thought we moved everyting out of the default namespace? in order to avoid issues where a customer could accidentally give a role permissions over default and get themselves in trouble

Reply:
@jiayiwang7 @danbudris 
What @jiayiwang7 says it's true, we install the resources from the spec file and the `Bundles` file in default
And all the other resources (capi, etc.) in `eksa-system`

I think `Bundles` should be in `eksa-system`, it's part of our "internal" resources and not something the user should be freely updating

As for the other eks-a resources, they should probably not be in `eksa-system`, since that's the namespace that should be "protected". If we put them there, we defeat the purpose of separating the "internal" resources from our public api

That said, I think we should support our "external" api objects to be created in other namespaces than default.
Whether we give the user the option to limit the namespaces where we watch from eksa objects or not, I think that's a separate conversation